nss-day-cohort-41/nutshell-the-last-rose-of-summer
# Messages section

Description

First Review.

Display/Create/Edit of messages implemented with most recent messages being displayed at the bottom of the list. The 'Add friend' button in the message display has a listener attached and is ready to be tied to functionality. The 'Add an Item' select element also has a listener attached. The logic is in place for all options, though only the 'Message' option has functionality right now.

There is a date converter function in the new file "miscSharedFunctions.js" that you can use to make 
=> new Date() <= work as a timestamp and display nicely

Type of change:

 **New feature (non-breaking change which adds functionality)

Testing Instructions
You may:
Add an Item => 'Message'
         Message field will display:
                 User Name (according to sessionStorage.activeUserName ****newly added, you will 
                         need to login for this to display anything besides 'undefined')
                 Message entry field
                 "Save Message" & "Discard Changes" buttons (both will clear the field DOM when 
                         clicked.) Save POSTs a new message to the database.

Edit a displayed message (Only available to the current user)
         Message field will display:
                 User Name (according to the saved user name attached to the message)
                 Message entry field
                 Date of last saved message
                 "Update Message" & "Discard Changes" buttons (both will clear the field DOM when 
                         clicked.) Update modifies the existing message and applies a fresh timestamp.

Clicking on the 'Add Friend' button will display a message on the console

Optional:
Place these over what you may have in the starter message JSON. This is indicative of the POST and PUT methods and will display properly from the start. You may also want to modify some "Following" keys to get a mixture of friends and strangers to see all functionality
{
      "userId": 0,
      "message": "Why is this called nutshell?",
      "date": "2020-06-26T17:58:57.776Z",
      "id": 1
    },
    {
      "userId": 1,
      "message": "Nuttybuddy1",
      "date": "2020-06-27T17:57:13.722Z",
      "id": 2
    },
    {
      "userId": 2,
      "message": "Nuttybuddy2 says hi",
      "date": "2020-06-28T18:31:39.122Z",
      "id": 3
    },

Checklist:
 My code follows the style guidelines of this project
 I have performed a self-review of my own code
 I have commented my code, particularly in hard-to-understand areas
 My changes generate no new warnings or errors
 I have added test instructions that prove my fix is effective or that my feature works
